Code/Framework/AzFramework/AzFramework/Viewport/ViewportScreen.h

```diff
@@ -112,9 +112,9 @@ namespace AzFramework
 
     //! Returns the aspect ratio of the viewport dimensions.
     //! @note Ensure a valid viewport size is provided to this function.
-    inline float AspectRatio(const AZ::Vector2& viewportSize)
+    inline float AspectRatio(const ScreenSize& viewportSize)
     {
-        AZ_Assert(viewportSize.GetY() > 0.0f, "AspectRatio called with invalid viewport size");
-        return viewportSize.GetX() / viewportSize.GetY();
+        AZ_Assert(viewportSize.m_height > 0.0f, "AspectRatio called with invalid viewport size");
+        return aznumeric_cast<float>(viewportSize.m_width) / aznumeric_cast<float>(viewportSize.m_height);
```

Comment:
Maybe make AspectRatio can be a member function of ScreenSize, seems useful there

Reply:
Hmm I quite like the fact `ScreenSize` is just a regular pod (and if we go with the change @mbalfour-amzn is suggesting we'd still need this free function...) but I see your point... Maybe we keep `WindowSize` and add this member function to that and I kill `ScreenSize` and then replace `ScreenPoint` and `ScreenVector` with `Vector2i`?

Reply:
I can try and make this change tomorrow too as I like this idea (haven't had time this week but should have a moment tomorrow 👍)

Reply:
This is now updated